packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -2068,3 +2130,406 @@ class RenderNestedScrollViewViewport extends RenderViewport {
+    _outerController.dispose();
+    _innerController.dispose();
+  }
+
+  @override
+  String toString() => '${objectRuntimeType(this, 'NestedScrollController')}(outer=$_outerController; inner=$_innerController)';
+}
+
+class _NestedScrollBehavior implements ScrollBehavior {
+  _NestedScrollBehavior(this.parentBehavior);
+
+  final ScrollBehavior parentBehavior;
+
+  @override
+  AndroidOverscrollIndicator get androidOverscrollIndicator => parentBehavior.androidOverscrollIndicator;
+
+  @override
+  Widget buildOverscrollIndicator(BuildContext context, Widget child, ScrollableDetails details) {
+    return parentBehavior.buildOverscrollIndicator(context, child, details);
+  }
+
+  /// Assign the innerScrollController as the default controller to scrollbar.
+  @override
+  Widget buildScrollbar(BuildContext context, Widget child, ScrollableDetails details) {
+    assert(details.controller is NestedScrollController);
+    final NestedScrollController controller = details.controller as NestedScrollController;
+    final ScrollableDetails nestedDetails = ScrollableDetails(direction: details.direction, controller: controller._innerController);
+    return parentBehavior.buildScrollbar(context, child, nestedDetails);
+  }
+
+  @override
+  Widget buildViewportChrome(BuildContext context, Widget child, AxisDirection axisDirection) {
+    return parentBehavior.buildViewportChrome(context, child, axisDirection);
+  }
+
+  @override
+  ScrollBehavior copyWith({
+    bool? scrollbars,
+    bool? overscroll,
+    ScrollPhysics? physics,
+    TargetPlatform? platform,
+    Set<PointerDeviceKind>? dragDevices,
+    AndroidOverscrollIndicator? androidOverscrollIndicator
+  }) {
+    return parentBehavior.copyWith(
+      scrollbars: scrollbars,
+      overscroll: overscroll,
+      dragDevices: dragDevices,
+      physics: physics,
+      platform: platform,
+      androidOverscrollIndicator: androidOverscrollIndicator
+    );
+  }
+
+  @override
+  Set<PointerDeviceKind> get dragDevices => parentBehavior.dragDevices;
+
+  @override
+  TargetPlatform getPlatform(BuildContext context) {
+    return parentBehavior.getPlatform(context);
+  }
+
+  @override
+  ScrollPhysics getScrollPhysics(BuildContext context) {
+    return parentBehavior.getScrollPhysics(context);
+  }
+
+  @override
+  bool shouldNotify(covariant _NestedScrollBehavior oldDelegate) {
+    return parentBehavior.shouldNotify(oldDelegate.parentBehavior);
+  }
+
+  @override
+  GestureVelocityTrackerBuilder velocityTrackerBuilder(BuildContext context) {
+    return parentBehavior.velocityTrackerBuilder(context);
+  }
+}
+
+/// A [NestedScrollController] whose [initialScrollOffset] tracks its most recently
+/// updated [ScrollPosition].
+///
+/// See also:
+///
+///  * [NestedScrollController], which this overrides.
+///  * [TrackingScrollController], whose initialScrollOffset tracks its most recently updated ScrollPosition.
+class TrackingNestedScrollController extends NestedScrollController {
```

Comment:
This is more confusing actually, there are three types of ScrollControllers? It does not look like anything is using this one, what is it for?

Reply:
It is for the test `Three NestedScrollViews with one ScrollController`. It is equivalent to `TrackingScrollController`. Is it better to remove this class from this PR?